chia/wallet/wallet_state_manager.py

```diff
@@ -797,6 +805,14 @@ async def determine_coin_type(
 
         uncurried = uncurry_puzzle(puzzle)
 
+        from chia.wallet.vault.vault_drivers import match_vault_puzzle
+
+        vault_check = match_vault_puzzle(uncurried.mod, uncurried.args)
+        if vault_check:
+            await self.handle_vault(puzzle, coin_spend, coin_state)
+            # Return None since we don't want to add the vault singleton to the normal wallet coin store
+            return None, None
```

Comment:
What's the reason for this?  I wouldn't assume adding the seen coin as a general coin record would hurt much and a surprising amount of our sync relies on the state of that table.